lib/models/package-info-cache.js

```diff
@@ -0,0 +1,1049 @@
+'use strict';
+
+/*
+ * Performance cache for information about packages (projects/addons/"apps"/modules)
+ * under an initial root directory and resolving addon/dependency links to other packages.
+ */
+
+const fs = require('fs-extra');
+const path = require('path');
+
+let realFilePathCache;
+let realDirectoryPathCache;
+
+/**
+ * Resolve the real path for a file, return null if does not
+ * exist or is not a file or FIFO, return the real path otherwise.
+ *
+ * @private
+ * @method getRealFilePath
+ * @param  {String} filePath the path to the file of interest
+ * @return {String} real path or null
+ */
+function getRealFilePath(filePath) {
```

Comment:
needs unit test

Reply:
How does one create a unit test for a private function, without creating an API that shouldn't be public?

Reply:
Exposed the function as part of module.exports, and added tests for getRealDirectoryPath and getRealFilePath.